benches/benchmarks.rs

```diff
@@ -74,6 +74,67 @@ pub fn benchmark(c: &mut Criterion) {
             },
         );
     }
+
+    {
+        let mut uncontended = c.benchmark_group("uncontended guarded");
+        uncontended.bench_function("read output", |b| b.iter(|| *output.output_buffer()));
+        uncontended.bench_function("clean update", |b| {
+            b.iter(|| {
+                output.update();
+            })
+        });
+        uncontended.bench_function("clean receive", |b| b.iter(|| *output.read()));
+        uncontended.bench_function("write input", |b| {
+            b.iter(|| {
+                *input.input_buffer_publisher() = black_box(0);
+            })
+        });
+        uncontended.bench_function("send", |b| {
+            b.iter(|| *input.input_buffer_publisher() = black_box(0))
+        });
+        uncontended.bench_function("transmit", |b| {
+            b.iter(|| {
+                *input.input_buffer_publisher() = black_box(0);
+                *output.read()
+            })
+        });
+    }
+
+    {
+        let (mut input, mut output) = TripleBuffer::<u8>::default().split();
+        let mut read_contended = c.benchmark_group("read contention guarded");
+        testbench::run_under_contention(
+            move || black_box(*output.read()),
+            || {
+                read_contended.bench_function("write input", |b| {
+                    b.iter(|| {
+                        *input.input_buffer_publisher() = black_box(0);
+                    })
+                });
+                read_contended.bench_function("send", |b| {
+                    b.iter(|| *input.input_buffer_publisher() = black_box(0))
+                });
+            },
+        );
```

Comment:
These benchmarks operate under the same contention workload as the existing "read contention" group, so I think they should be new benchmarks under the "read contention" existing group. To avoid name collisions and redundant benchmarks, you will want to rename "write input" to "guarded write+send", and as before delete "send" which is just a copy of "write input".